CORE-5045 Register custom internal AMQP serializers in flow worker

lankydan opened the pull request:
Call `AMQPSerializationScheme.registerCustomSerializers` in
`FlowSandboxServiceImpl` as it was incorrectly trying to scan for
serializers which are not currently OSGi components and
`ThrowableSerializer` couldn't be converted as it had a constructor
argument.

Remove now unused code from `EntitySandboxServiceImpl` and
`FlowSandboxServiceImpl`.

barrylapthorn reviewed (dismissed):
Seems reasonable.

chrisr3 commented:
This begs the question: How are you discovering custom serializers if they are no longer OSGi components? Have you just abandoned `net.corda.crypto.impl.serialization.PublicKeySerializer`?

The idea was that any bundle could define its own `InternalCustomSerializer` services to handle its own types.

lankydan commented:
Will change this PR after talking with @chrisr3.

lankydan committed:
CORE-5045 Register custom internal AMQP serializers in flow worker

Call `AMQPSerializationScheme.registerCustomSerializers` in
`FlowSandboxServiceImpl` as it was incorrectly trying to scan for
serializers which are not currently OSGi components and
`ThrowableSerializer` couldn't be converted as it had a constructor
argument.

lankydan commented:
@barrylapthorn I've removed the db worker changes, and now have the flow worker adding the missing line (which the db worker already had).

chrisr3 reviewed (approved):
LGTM, registering the basic ones before the discovered ones :grinning:.
